core/server/master/src/main/java/alluxio/master/metastore/ReadOnlyInodeStore.java

```diff
@@ -31,22 +38,125 @@
    */
   Optional<Inode> get(long id);
 
+  /**
+   * Returns an iterable for the ids of the children of the given directory.
+   *
+   * @param inodeId an inode id to list child ids for
+   * @return the child ids iterable
+   */
+  Iterable<Long> getChildIds(Long inodeId);
+
+  /**
+   * Returns an iterable for the ids of the children of the given directory.
+   *
+   * @param inode the inode to list child ids for
+   * @return the child ids iterable
+   */
+  default Iterable<Long> getChildIds(InodeDirectoryView inode) {
+    return getChildIds(inode.getId());
+  }
+
+  /**
+   * Returns an iterator over the children of the specified inode.
+   *
+   * The iterator is weakly consistent. It can operate in the presence of concurrent modification,
+   * but it is undefined whether concurrently removed inodes will be excluded or whether
+   * concurrently added inodes will be included.
+   *
+   * @param inodeId an inode id
+   * @return an iterable over the children of the inode with the given id
+   */
+  default Iterable<? extends Inode> getChildren(Long inodeId) {
+    return () -> {
+      Iterator<Long> it = getChildIds(inodeId).iterator();
+      return new Iterator<Inode>() {
+        private Inode mNext = null;
+
+        @Override
+        public boolean hasNext() {
+          advance();
+          return mNext != null;
+        }
+
+        @Override
+        public Inode next() {
+          if (!hasNext()) {
+            throw new NoSuchElementException("No more inodes");
```

Comment:
NIT: would something like `"no more children in iterator for inode id " + inodeId`